File: 1684.count-the-number-of-consistent-strings.java
/*
 * @lc app=leetcode id=1684 lang=java
 *
 * [1684] Count the Number of Consistent Strings
 */

// @lc code=start
class Solution {
    public int countConsistentStrings(String allowed, String[] words) {
     HashSet<Character> allw = new HashSet<>();
        for (char c : allowed.toCharArray()) {
            allw.add(c);
        }
        int count = 0;
        for (String s : words) {
            for (char c : s.toCharArray()) {
                if (!allw.contains(c)) {
                    count++;
                    break;
                }
            }
        }
        return words.length - count;   
    }
}
// @lc code=end

File: 1684.count-the-number-of-consistent-strings.py
#
# @lc app=leetcode id=1684 lang=python
#
# [1684] Count the Number of Consistent Strings
#

# @lc code=start
class Solution(object):
    def countConsistentStrings(self, allowed, words):
        """
        :type allowed: str
        :type words: List[str]
        :rtype: int
        """
        allw = set(allowed)
        c = 0
        for i in words:
            for j in i:
                if j not in allw:
                    c += 1
                    break
        return len(words) - c
        
# @lc code=end

